lib/containers/engine.pm

```diff
@@ -0,0 +1,396 @@
+    return $self->_engine_script_run("pull $image_name", timeout => $args{timeout} // 300);
+}
+
+=head2 commit
+
+Save a existing container as a new image in the local registry
+
+=cut
+sub commit {
+    my ($self, $mycontainer, $new_image_name, %args) = @_;
+    $self->_engine_assert_script_run("commit $mycontainer $new_image_name", timeout => $args{timeout});
+}
+
+=head2 enum_images
+
+Return an array ref of the images
+
+=cut
+sub enum_images {
+    my ($self) = shift;
+    my $images_s = $self->_engine_script_output("images -q");
+    record_info "Images", $images_s;
+    my @images = split /[\n\t]/, $images_s;
+    return \@images;
+}
+
+=head2 enum_images
+
+Return an array ref of the containers
+
+=cut
+sub enum_containers {
+    my ($self) = shift;
+    my $containers_s = $self->_engine_script_output("container ls -q");
+    record_info "Containers", $containers_s;
+    my @containers = split /[\n\t]/, $containers_s;
+    return \@containers;
+}
+
+=head2 get_images_by_repo_name
+
+Returns an array ref with the names of the images.
+
+=cut
+sub get_images_by_repo_name {
+    my ($self)      = @_;
+    my $repo_images = $self->_engine_script_output("images --format '{{.Repository}}'", timeout => 60);
+    my @images      = split /[\n\t ]/, $repo_images;
+    return \@images;
+}
+
+=head2 info
+
+Assert a C<property> against given expected C<value> if C<value> is given.
+Otherwise it prints the output of info.
+
+=cut
+sub info {
+    my ($self, %args) = shift;
+    my $property = $args{property} ? qq(--format '{{.$args{property}}}') : '';
+    my $expected = $args{value}    ? qq( | grep $args{value})            : '';
+    $self->_engine_assert_script_run(sprintf("info %s %s", $property, $expected));
+}
+
+=head2 get_container_logs($container)
+
+Request container's logs.
+C<container> the running container.
+C<logs> returns a string.
+
+=cut
+sub get_container_logs {
+    my ($self, $container) = @_;
+    my $logs = $self->_engine_script_output("container logs $container");
+    return $logs;
+}
+
+=head2 remove_image($image_name)
+
+Remove a image from the pool.
+
+=cut
+sub remove_image {
+    my ($self, $image_name) = @_;
+    $self->_engine_assert_script_run("rmi -f $image_name");
+}
+
+=head2 remove_container
+
+Remove a container from the pool.
+
+=cut
+sub remove_container {
+    my ($self, $container_name) = @_;
+    $self->_engine_assert_script_run("rm -f $container_name");
+}
+
+=head2 check_image_in_host
+
+Returns true if host contains C<img> or false.
+
+=cut
+sub check_image_in_host {
+    my ($self, $img) = @_;
+    my @lregistry = $self->enum_images();
+    $img =~ @lregistry;
+}
+
+=head2 configure_insecure_registries
+
+Updates the registry files for the running container runtime to allow access to
+insecure registries.
+
+Implementation is subject to the the subclass otherwise call to this subroutine dies.
+
+=cut
+sub configure_insecure_registries {
+    my ($self) = shift;
```

Comment:
```suggestion
    return
```

Reply:
i believe `$self` is better to stay here. When `$instance->configure_insecure_registries` will not find the method. i am leaving as it is.

Reply:
Whatever, but it is basically the same if the method is not implemented by the sub-class. 
see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L80

Reply:
*this change is ready for review*